### PR TITLE
Example k8s job to setup buckets using STS and mc

### PR DESCRIPTION
**Pull request by pjuarezd**
To provide an example on how to use `Operator STS` with `mc` client.

### CODE REVIEW

**Comment by pjuarezd** [comment]
This example depends on https://github.com/minio/mc/pull/4744 to be merged and released

**Review by dvaldivia — DISMISSED** [review]

File: examples/kustomization/sts-example/sample-data/mc-job-setup-bucket.yaml [hunk 1]
```diff
@@ -35,7 +35,8 @@ spec:
   backoffLimit: 5
   template:
     spec:
-      restartPolicy: OnFailure
+      serviceAccountName: mc-job-sa
+      restartPolicy: Never
```
Comment (jiuker) [inline examples/kustomization/sts-example/sample-data/mc-job-setup-bucket.yaml:39]: If Job and tenant are created at the same time, it won't execute all the time here, is this expected?

**Review by jiuker — CHANGES_REQUESTED** [review]

**Comment by pjuarezd** [comment]
Waiting for a mc release including PR https://github.com/minio/mc/pull/4754 to pass failing test

File: examples/kustomization/sts-example/sample-data/mc-job-setup-bucket.yaml (hunk 1 above)
Comment (pjuarezd) [inline examples/kustomization/sts-example/sample-data/mc-job-setup-bucket.yaml:39]:
```suggestion
      restartPolicy: OnFailure
```

Comment (pjuarezd) [reply examples/kustomization/sts-example/sample-data/mc-job-setup-bucket.yaml]: that's rght, having `restartPolicy: OnFailure` will retry Job if tenant not ready yet. This was working before.

**Review by jiuker — APPROVED** [review]

**Review by dvaldivia — APPROVED** [review]

**Pull request merged**